### DISCUSSION TITLE
How to operate on all of entities in a akka clustering shard region

### DISCUSSION THREAD

**Discussion by alireza267** [akkadotnet/akka.net#4753] [Q&A]
- v1.4.11
- DotNet5

Is there a way to get the list of active entities(actors) in a shard region?
Is there a way to broadcast to all of the entities in a shard region?

**Comment by Horusiath** [comment]
While all of the commands are in the docs, I think the most readable overview of them can be found at the bottom of [this article](https://petabridge.com/blog/cluster-sharding-technical-overview-akkadotnet/).

**Reply by alireza267** [reply]
I found that GetShardRegionState could get all identifiers of the active entities in a shard region. despite a shard region with multi millions of entities, the response message CurrentShardRegionState is very large. Is it an issue? big response size.

**Comment by alireza267** [comment]
thanks @Horusiath. but I hoped there is simpler solution for operations like broadcast on a shardRegion. anyway this could help to.
